akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/ClusterShardingSpec.scala

```diff
@@ -150,18 +150,66 @@ class ClusterShardingSpec extends TestKit("ClusterShardingSpec", ClusterSharding
 
   val typeKey2 = EntityTypeKey[IdTestProtocol]("no-envelope-shard")
   val behaviorWithId = Behaviors.immutable[IdTestProtocol] {
-    case (_, IdStopPlz(_)) ⇒
+    case (_, IdStopPlz()) ⇒
       Behaviors.stopped
 
     case (ctx, IdWhoAreYou(_, replyTo)) ⇒
-      replyTo ! s"I'm ${ctx.self.path.name}"
+      val address = Cluster(ctx.system).selfMember.address
+      replyTo ! s"I'm ${ctx.self.path.name} at ${address.host.get}:${address.port.get}"
       Behaviors.same
 
     case (_, IdReplyPlz(_, toMe)) ⇒
       toMe ! "Hello!"
       Behaviors.same
   }
 
+  val shardingRef1: ActorRef[ShardingEnvelope[TestProtocol]] = sharding.spawn[TestProtocol](
+    _ ⇒ behavior,
+    Props.empty,
+    typeKey,
+    ClusterShardingSettings(system),
+    10,
+    StopPlz())
+
+  val shardingRef2 = sharding2.spawn[TestProtocol](
```

Comment:
with the `entityId => Behavior` parameter some type inference is lost and `TestProtocol` must be defined like that